.github/workflows/build.yml

```diff
@@ -0,0 +1,326 @@
+          while [ $isDone = false ]
+          do
+            result=$(sshpass -p ${{ env.vmAdminPassword }} -v ssh -p 22 -o StrictHostKeyChecking=no -o ConnectTimeout=100 -v -tt ${{ env.vmAdminId }}@${entitledVMIP} '(tail -n1) </var/log/cloud-init-was.log')
+            # Remove special characters
+            result=$(echo $result | sed $'s/[^[:alnum:]\t]//g')
+            if [[ "$result" == "Unentitled" ]] || [[ "$result" == "Entitled" ]]; then
+                isDone=true
+            else
+                echo "waiting for was entitlement check completed..."
+                sleep 5
+            fi
+          done
+          echo $result
+
+          if [ ${result} = Unentitled ]; then
+              exit 1
+          fi
+      - name: Delete all resources but vhd storage account in the test resource group
+        id: delete-resources-in-group
+        if: always()
+        uses: azure/CLI@v1
+        with:
+          azcliversion: ${{ env.azCliVersion }}
+          inlineScript: |
+            # Disks have to be deleted after the VM is deleted
+            unentitledOsDiskId=$(az vm show --resource-group ${{ env.testResourceGroup }} --name untitled${{ github.run_id }}${{ github.run_number }} --query "storageProfile.osDisk.managedDisk.id" -o tsv)
+            echo "unentitledOsDiskId: ${unentitledOsDiskId}"
+
+            unentitledDataDiskIds=$(az vm show --resource-group ${{ env.testResourceGroup }} --name untitled${{ github.run_id }}${{ github.run_number }} --query "storageProfile.dataDisks" | jq -r 'map(.managedDisk.id) | join(" ")')
+            echo "unentitledDataDiskIds: ${unentitledDataDiskIds}"
+
+            entitledOsDiskId=$(az vm show --resource-group ${{ env.testResourceGroup }} --name entitled${{ github.run_id }}${{ github.run_number }} --query "storageProfile.osDisk.managedDisk.id" -o tsv)
+            echo "unentitledOsDiskId: ${unentitledOsDiskId}"
+
+            entitledDataDiskIds=$(az vm show --resource-group ${{ env.testResourceGroup }} --name entitled${{ github.run_id }}${{ github.run_number }} --query "storageProfile.dataDisks" | jq -r 'map(.managedDisk.id) | join(" ")')
+            echo "unentitledDataDiskIds: ${unentitledDataDiskIds}"
+
+            resourcesToDelete=$(az resource list --query "[?name!='${{ env.vhdStorageAccountName }}']" --resource-group ${{ env.testResourceGroup }} | jq -r 'map(.id) | join(" ")')
+            echo $resourcesToDelete
+
+            az resource delete --verbose --ids $resourcesToDelete
+            az disk delete --yes --resource-group ${{ env.testResourceGroup }} --ids ${unentitledOsDiskId} ${unentitledDataDiskIds} ${entitledOsDiskId} ${entitledDataDiskIds}
+      - name: Generate SAS url
+        id: generate-sas-blob-url
+        run: |
+          #Get a minus-24-hour date for the SAS token
+          minus24HoursUtc=$(date -u --date "$dte -24 hour" +%Y-%m-%dT%H:%MZ)
+          echo $minus24HoursUtc
+
+          #Get a plus-30-day date for the SAS token
```

Comment:
What is the relationship between 30 days token and partner center VM image?

Reply:
Explained in the CR meeting. Will need further exploration on Partner center's rules

Reply:
I assert we can commit this without having the full explanation because we have changed the trigger to be using CURL.